### ISSUE TITLE
process not restarted on failure

### ISSUE THREAD

**Issue by joprice** [bazelbuild/bazel-watcher#385]
I'm using ibazel with and when compilation fails, I intermittently get errors like `Error starting process: fork/exec` ` Run start failed fork/exec...permission denied`, followed by `Error writing success to stdin: write |1: file already closed`. After restarting the process, things start working again. Is this an issue with ibazel not restarting the underlying process?

**Comment by booninite** [comment]
I started running into this only after I started using `.bazel_fix_commands.json`, not sure if it is related.  I assumed that it had something to do with ownership of the `bazel` binary, or something.  

For additional context, my `bazel` binary is actually `bazelisk`.

**Comment by joprice** [comment]
I'm not using bazelisk so it's likely unrelated to that.

**Comment by achew22** [comment]
@joprice and @booninite, thanks for reporting the issue. I think the issue you're running into is the same as https://github.com/bazelbuild/bazel-watcher/issues/300. In order to fix this, I created https://github.com/bazelbuild/bazel-watcher/pull/378 which adds an example client (an inner program) that has a http interface that helps with testing. On this http interface I added a method that lets you remotely terminate the server.

It works great and I have even been able to fix this bug in e2e tests, but I haven't yet been able to get it to pass on Windows, We're close on it, I just don't have a windows box to do devwork on so I have a 10 minute latency on all my tests which inevitably leads me to working on something else and then not returning. If either of you are knowledgable about windows runfiles I would really appreciate some help.
